### PR TITLE
Change logging format

### PR DESCRIPTION
**Pull request by nabuskey**
Currently, log outputs are hard to read. For example:

```
time=2024-08-26T20:22:19.468Z level=INFO msg="Creating kind cluster" logger=setup cluster=localdev
time=2024-08-26T20:22:36.409Z level=INFO msg="Done creating cluster" logger=setup cluster=localdev
time=2024-08-26T20:22:36.566Z level=INFO msg="Adding CRDs to the cluster" logger=setup
time=2024-08-26T20:22:37.377Z level=INFO msg="Setting up CoreDNS" logger=setup
time=2024-08-26T20:22:37.404Z level=INFO msg="Setting up TLS certificate" logger=setup
time=2024-08-26T20:22:37.425Z level=INFO msg="Creating localbuild resource" logger=setup
time=2024-08-26T20:22:37.425Z level=INFO msg="Starting EventSource" controller=localbuild controllerGroup=idpbuilder.cnoe.io controllerKind=Localbuild source="kind source: *v1alpha1.Localbuild"
```

There is a lot of repeated texts that adds no value. 

With this PR log messages look like:
```
Aug 27 19:45:47 INFO Creating kind cluster logger=setup
Aug 27 19:45:47 INFO Runtime detected logger=setup provider=docker
Aug 27 19:45:47 INFO Cluster already exists logger=setup cluster=localdev
hi
Aug 27 19:45:47 INFO Adding CRDs to the cluster logger=setup
Aug 27 19:45:47 INFO Setting up CoreDNS logger=setup
Aug 27 19:45:47 INFO Setting up TLS certificate logger=setup
Aug 27 19:45:47 INFO Creating localbuild resource logger=setup
Aug 27 19:45:47 INFO Starting EventSource controller=localbuild controllerGroup=idpbuilder.cnoe.io controllerKind=Localbuild source=kind source: *v1alpha1.Localbuild
```

This also adds colored output option. Disabled by default.

image

### CODE REVIEW

**Comment by nabuskey** [comment]
> Don't forget to document how to enable/disable the colored output ;-)

Yep. Plan is to make an issue to track it if this gets merged :)

**Comment by tapas4java** [comment]
Any problem if we make colored output default to `true` ? Anyway  it's not going to break anything

**Comment by nabuskey** [comment]
Coloring is very subjective and I didn't want to impose what I like to others by default. If others like it over time, we can make it enabled by default imo.

File: pkg/cmd/root.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ var rootCmd = &cobra.Command{
 
 func init() {
 	rootCmd.PersistentFlags().StringVarP(&helpers.LogLevel, "log-level", "l", "info", helpers.LogLevelMsg)
+	rootCmd.PersistentFlags().BoolVarP(&helpers.ColoredOutput, "color", "C", false, helpers.ColoredOutputMsg)
```
Comment (nimakaviani) [inline pkg/cmd/root.go:23]: I feel like I am ok with not having a short form flag for color. it is a luxury now, so doing `--color` should be just fine. We dont have any other capital short forms for flags, and if we want to introduce one, probably better to do it for an actual feature.

Comment (nabuskey) [reply pkg/cmd/root.go:23]: I'm ok with that.

**Review by nimakaviani — APPROVED** [review]
LGTM

**Pull request merged**